mobile-widgets/qmlprefs.cpp

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "qmlprefs.h"
+#include "qmlmanager.h"
+
+#include "core/membuffer.h"
+#include "core/subsurface-qt/SettingsObjectWrapper.h"
+#include "core/gpslocation.h"
+
+
+/*** Global and constructors ***/
+QMLPrefs *QMLPrefs::m_instance = NULL;
+
+QMLPrefs::QMLPrefs() :
+	m_cloudPassword(""),
+	m_cloudPin(""),
+	m_cloudUserName(""),
+	m_credentialStatus(CS_UNKNOWN),
+	m_developer(false),
+	m_distanceThreshold(1000),
+	m_oldStatus(CS_UNKNOWN),
+	m_showPin(false),
+	m_timeThreshold(60)
+{
+	if (!m_instance)
+		m_instance = this;
```

Comment:
This is very strange. I know it we use something like this for MainWindow and QMLManager, but these are explicitly constructed/destructed. I would prefer not having more of this strange pattern.

Does this even have to be a global object? If yes, why not use the classical construct-on-demand pattern?

Reply:
I added a comment in the code. I would love to avoid this strange construct, but QMLEngine makes a new QMLPrefs and it does need to be a global object, at least for now, otherwise the variables may have different values.

I have searched for a pattern, where the constructor is called (with new ...) but the existing this pointer is returned, in practice make a singleton for several new calls.

Suggestions are very welcome.

Reply:
we ended up with this pattern because otherwise we weren't able to access the object from C++. Somehow the QML code instantiated its own copy of the class and we were forever trying to debug why changes that we made in C++ weren't reflected in QML

Reply:
Exactly what I experienced, the only sane way to avoid an ugly constructor is to use a interface/implementation pattern, which we have 98% in using pref.h. I wanted first to showcase that a separation from QMLManager works (this PR), and then do the full implementation.